src/scripts/filter_rare_disease_list.py

```diff
@@ -0,0 +1,26 @@
+import argparse
+import pandas as pd
+
+def main(args):
+    # Read the input files into pandas dataframes
+    old_df = pd.read_csv(args.old_rare_list, sep='\t')
+    new_df = pd.read_csv(args.new_rare_list, sep='\t')
+
+    # Filter the dataframes
+    df_added = old_df[~old_df['ID'].isin(new_df['ID'])]
+    df_removed = new_df[~new_df['ID'].isin(old_df['ID'])]
+
+    # Save the filtered dataframes to TSV files
+    df_removed.to_csv(args.new_rare_diseases, sep='\t', index=False)
+    df_added.to_csv(args.lost_rare_diseases, sep='\t', index=False)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Process and filter two TSV files")
+    parser.add_argument("old_rare_list", help="Path to the first input TSV file")
+    parser.add_argument("new_rare_list", help="Path to the second input TSV file")
+    parser.add_argument("new_rare_diseases", help="Path to the first output TSV file")
+    parser.add_argument("lost_rare_diseases", help="Path to the second output TSV file")
+
+    args = parser.parse_args()
+
+    main(args)
```

Comment:
### Argparse style
I use a few different patterns but I prefer `main(**vars(parser.parse_args()))` to `main(args)`.

This will make it so that you are passing normal keyword arguments to `main()`. That way, you can define your parameters in `main()` and you can also call it directly rather than just through the CLI.

It's not a big deal though. What's here is not very verbose, which is also advantageous. If you want you can compare to [migrate](https://github.com/monarch-initiative/mondo-ingest/blob/bugfix-omimps-labels/src/scripts/migrate.py), which has some other differences.

Reply:
TBH, I asked ChatGPT to generate this script and didn't even look at it much. I don't like Argparse so much anymore, I prefer click. But for a small ad-hoc script like this where I i just didn't know how to do it outside python, its fine. Thanl you for the feedback though! Very interesting!